base/abstractarray.jl

```diff
@@ -62,7 +62,11 @@ ndims{T<:AbstractArray}(::Type{T}) = ndims(super(T))
 length(t::AbstractArray) = prod(size(t))::Int
 endof(a::AbstractArray) = length(a)
 first(a::AbstractArray) = a[1]
-first(a) = (s = start(a); done(a, s) ? throw(ArgumentError("collection must be non-empty")) : next(a, s)[1])
+function first(arr)
```

Comment:
That's fixed now, you're right.
To me, the `a` also strongly implied it was for arrays, so I do think `itr` is clearer,
and given that I've spent most all of my time looking at string handle code, where `s` reads as `string` and not `state`, you can see why I like a little clearer names.
(I know *you* know exactly what you meant, but the idea is to make things clearer for somebody new to this part of the code)